arelle/formula/FactAspectsCache.py

```diff
@@ -10,24 +10,60 @@
 
 
 class FactAspectsCache:
-    def __init__(self) -> None:
-        self._matchingAspects: defaultdict[ModelFact, defaultdict[ModelFact, defaultdict[int | QName, bool | None]]] = defaultdict(lambda: defaultdict(lambda: defaultdict(lambda: None)))
+    def __init__(self, maxSize: int) -> None:
+        self._size = 0
+        self._maxSize = maxSize if maxSize >= 0 else float("inf")
+        self._maxSizePreviouslyReached = False
+        self._prioritizedAspects: set[int | QName] = set()
+        self._matchingAspects: defaultdict[
```

Comment:
pre-existing: `defaultdict` is an implementation detail, it seems more accurate to type as `dict[ModelFact, dict[ModelFact, dict[int | QName, bool | None]]]`

Reply:
defaultdict is guaranteed to return a value even if the key doesn't exist. That isn't an implementation detail. FormulaEvaluator depends on that behavior (instead of using `dict.get()` which turned out to be slower)